libyamlscript/src/libyamlscript/API.java

```diff
@@ -16,9 +16,10 @@ public final class API {
     ) {
         String ys = CTypeConversion.toJavaString(s);
         String json = libyamlscript.core.evalYsToJson(ys);
-        CTypeConversion.CCharPointerHolder holder = CTypeConversion.toCString(json);
-        CCharPointer value = holder.get();
-        return value;
+        try (CTypeConversion.CCharPointerHolder holder = CTypeConversion.toCString(json)) {
```

Comment:
The documentation says:
> An auto-closable that holds a Java [CharSequence](http://docs.oracle.com/javase/21/docs/api/java/lang/CharSequence.html?is-external=true) as a null-terminated C char[] array. The C pointer is only valid as long as the auto-closeable has not been closed.

I would be very very surprised this way the way to go about it...

Reply:
I have come to the same conclusion. I know not how nor why it works. I know very little Java, so I'm open to a better suggestion.
The aim here is to remove the memory leak. Prior to that addition, any string resulting from `toCString` would never be freed and leak.
Using the `try` is the method I found in every blog/tutorial I could find. I share your feeling that this does not _seem_ right.

I have however implemented that, and written an example Rust program that uses that specific code to generate a million strings from Java, effectively calling this a million times. The string that was returned was read in the Rust code, then copied, then discarded (so as to not hoard memory on the Rust side either). The string was successfully read and copied. Running that test program under `valgrind` lead to no memory error either.

I have expressed my doubts to @ingydotnet about the validity of this solution. I can only say that it works in practice, but I know not why.

Reply:
Here are some examples of places where I found that code:
  - https://stackoverflow.com/a/64089918
  - https://gist.github.com/komamitsu/43a9f3754c32236adc3d824e0624eb48